ui/app/src/components/VariableList/VariablesList.tsx

```diff
@@ -23,44 +23,47 @@ import { useSnackbar } from '@perses-dev/components';
 import Clipboard from 'mdi-material-ui/ClipboardOutline';
 import { useIsReadonly } from '../../model/config-client';
 import { DeleteVariableDialog } from '../dialogs';
-import { useUpdateVariableMutation } from '../../model/project-client';
 import { VariableDataGrid, Row } from './VariableDataGrid';
 import { VariableFormDrawer } from './VariableFormDrawer';
 
-export interface VariableListProperties {
-  projectName: string;
-  variableList: VariableResource[];
+type DispatchWithPromise<A> = (value: A) => Promise<void>;
+
+export interface VariableListProperties<T extends Variable> {
+  data: T[];
   hideToolbar?: boolean;
+  onUpdate: DispatchWithPromise<T>;
+  onDelete: DispatchWithPromise<T>;
   initialState?: GridInitialStateCommunity;
   isLoading?: boolean;
 }
 
 /**
  * Display variables in a table style.
- * @param props.variableList Contains all variables to display
+ * @param props.data Contains all variables to display
  * @param props.hideToolbar Hide toolbar if enabled
+ * @param props.onUpdate Event received when an update action has been requested
```

Comment:
I didn’t try it but curious, can a GlobalVariable be used inside a ProjectVariable?

Reply:
I think yes, but nothing is checked when you create a project variable. The same question for the dependency between project variables. 

That's an interesting point that we need to clarify for us, and in the view as well.